### ISSUE TITLE
clusterer.prediction_data_.cluster_tree.shape[0] = 0 in case of one cluster

### ISSUE THREAD

**Issue by divyagupta25** [scikit-learn-contrib/hdbscan#313]
Hi,

I am using hdbscan in the following manner, and getting a single cluster for my data. 

clusterer = hdbscan.HDBSCAN(min_cluster_size=50, min_samples=40, allow_single_cluster=True, prediction_data=True)
cluster_labels = clusterer.fit_predict(data_df)
n_clusters = len(set(cluster_labels)) - (1 if -1 in cluster_labels else 0)     # gives 1

However, I am unable to use approximate_predict() on new data points as - 
print(clusterer.prediction_data_.cluster_tree.shape[0])                          # gives 0

Hence, it reports the warning - 
/usr/lib64/python2.7/site-packages/hdbscan/prediction.py:385: UserWarning: Clusterer does not have any defined clusters, new data will be automatically predicted as noise.

But one cluster is present. 

Any help on this would be appreciated.
Thanks in advance!

**Comment by lmcinnes** [comment]
Yes, this is technically "correct". The single cluster case is kind of an odd exceptional case so various things don't quite work out. The ``cluster_tree`` in the ``prediction_data`` is really just the condensed tree with all the information about points falling out of clusters (i.e. splits below the ``min_cluster_size``) removed. As it happens since there is only one cluster that decays there is no information left. If you like you can imagine that there is essentially an implicit entry of a root cluster creation at ``lambda == 0``.
